src/trace_processor/perfetto_sql/stdlib/wattson/tasks/threads_w_processes.sql

```diff
@@ -38,3 +40,120 @@ LEFT JOIN android_process_metadata AS package
   USING (upid)
 WHERE
   dur > 0;
+
+-- Gets all IRQs and labels each slice with hard or soft IRQ
+CREATE PERFETTO TABLE _all_irq_slices AS
+SELECT
+  id,
+  ts,
+  dur,
+  parent_id,
+  extract_arg(arg_set_id, 'vec') IS NULL AS is_hard_irq
+FROM slice
+WHERE
+  category = 'irq';
```

Comment:
This is not a good way to filter for irqs, it's very possible for other people to emit events with a category as generic as "irq".

Suggest instead joining with track and looking for track.type `cpu_softirq` and `cpu_irq`